weave/tests/test_evaluations.py

```diff
@@ -144,3 +152,366 @@ def is_op_ref_with_name(val: Any, name: str):
     assert is_op_ref_with_name(expected_predict_ref, "MyModel.predict")
+
+
+@weave.op
+def gpt_mocker(question: str):
+    return {
+        "response": str(question),
+        "model": "gpt-4o-2024-05-13",
+        "usage": {
+            "requests": 1,
+            "completion_tokens": 28,
+            "prompt_tokens": 11,
+            "total_tokens": 39,
+        },
+    }
+
+
+class SimpleModel(Model):
+    @weave.op()
+    def predict(self, question: str):
+        res = gpt_mocker(question.strip(".") if len(question) % 2 == 1 else question)
+        return {"response": res["response"]}
+
+
+class SimpleModelWithConfidence(Model):
+    @weave.op()
+    def predict(self, question: str):
+        res = gpt_mocker(question.strip(".") if len(question) % 2 == 0 else question)
+        return {"response": res["response"], "confidence": 1 / (len(res) + 1)}
+
+
+def score_int(expected: str, model_output: dict) -> int:
+    matches = 0
+    for i in range(min(len(expected), len(model_output["response"]))):
+        if expected[i] == model_output["response"][i]:
+            matches += 1
+    return matches
+
+
+def score_float(expected: str, model_output: dict) -> float:
+    matches = score_int(expected, model_output)
+    return matches / max(len(expected), len(model_output["response"]))
+
+
+def score_bool(expected: str, model_output: dict) -> bool:
+    return score_float(expected, model_output) == 1.0
+
+
+def score_dict(expected: str, model_output: dict) -> dict:
+    return {
+        "d_int": score_int(expected, model_output),
+        "d_float": score_float(expected, model_output),
+        "d_bool": score_bool(expected, model_output),
+        "d_nested": {
+            "d_int": score_int(expected, model_output),
+            "d_float": score_float(expected, model_output),
+            "d_bool": score_bool(expected, model_output),
+        },
+        "reason": "This is a test reason",
+    }
+
+
+class MyIntScorer(weave.Scorer):
+    @weave.op()
+    def score(self, expected: str, model_output: dict) -> int:
+        return score_int(expected, model_output)
+
+
+class MyFloatScorer(weave.Scorer):
+    @weave.op()
+    def score(self, expected: str, model_output: dict) -> float:
+        return score_float(expected, model_output)
+
+
+class MyBoolScorer(weave.Scorer):
+    @weave.op()
+    def score(self, expected: str, model_output: dict) -> bool:
+        return score_bool(expected, model_output)
+
+
+class MyDictScorer(weave.Scorer):
+    @weave.op()
+    def score(self, expected: str, model_output: dict) -> dict:
+        return score_dict(expected, model_output)
+
+
+class MyDictScorerWithCustomFloatSummary(weave.Scorer):
+    @weave.op()
+    def score(self, expected: str, model_output: dict) -> dict:
+        return score_dict(expected, model_output)
+
+    @weave.op()
+    def summarize(self, score_rows: list) -> Optional[dict]:
+        float_avg = sum([row["d_float"] for row in score_rows]) / len(score_rows)
```

Comment:
Can remove square brackets here inside sum (and similar cases below)

Reply:
👍 in next commit